### ISSUE TITLE
CondaHTTPError: HTTP None None for url <https://repo.continuum.io/pkgs/free/win-64/repodata.json.bz2><br>Elapsed: None<br><br>An HTTP error occurred when trying to retrieve this URL.<br>HTTP errors are often intermittent, and a simple retry will get you on your way.<br>ConnectTimeout(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/free/win-64/repodata.json.bz2 (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x0000000005E606A0>, 'Connection to repo.continuum.io timed out. (connect timeout=9.15)'))",),)<br>Navigator Error

### ISSUE THREAD

**Issue by Ropar** [ContinuumIO/anaconda-issues#2760]
## Main error
The following errors occurred:
## Traceback
```
CondaHTTPError: HTTP None None for url <https://repo.continuum.io/pkgs/free/win-64/repodata.json.bz2>
Elapsed: None

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectTimeout(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/free/win-64/repodata.json.bz2 (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x0000000005E606A0>, 'Connection to repo.continuum.io timed out. (connect timeout=9.15)'))",),)

```
## System information
```
python: 3.6.1
language: en_US
os: Windows;7;6.1.7601;AMD64;Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
version: 1.6.2
platform: win-64
qt: 5.6.2
pyqt: 5.6.0
conda: 4.3.21
```

**Comment by goanpeca** [comment]
Closing as duplicate of #1932

Please update to the latest version of Navigator.

On Navigator click on the update button on the top right of the interface or
on the terminal type

```
$ conda update anaconda-navigator
```

**Comment by gstroudfe** [comment]
Hallo. I receive exactly the same error, which is different from the one of issue 1932. The problem is that I cannot update anything. Any command results in this error.  

Fetching package metadata ...
CondaHTTPError: HTTP None None for url <https://repo.continuum.io/pkgs/free/win-64/repodata.json.bz2>
Elapsed: None

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectTimeout(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/free/win-64/repodata.json.bz2 (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x00000248C594E2B0>, 'Connection to repo.continuum.io timed out. (connect timeout=9.15)'))",),)

**Comment by rimiprasad** [comment]
@gstroudfe: Did you get any help ? I am facing issue

**Comment by gstroudfe** [comment]
@rimiprasad Unfortunately not. I just did not use Anaconda.

**Comment by ejahanpour** [comment]
@gstroudfe and @rimiprasad I tried disabling my ssl verification and it worked for me. Not sure is this is the end solution though. The command to do so: `conda config --set ssl_verify no`

**Comment by gnomic** [comment]
I'm doing conda update conda and getting
`Solving environment: failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.continuum.io/pkgs/r/noarch/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ProxyError(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/r/noarch/repodata.json.bz2 (Ca
used by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000000006169438>: Failed t
o establish a new connection: [Errno 11004] getaddrinfo failed',)))",),)`

This was working. The .condarc file is valid with ssl_verify: false and valid proxy.

Please advise

**Comment by sowmiyanm** [comment]
`conda update navigator-updater`
Fetching package metadata ...
CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.continuum.io/pkgs/main/osx-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/main/osx-64/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x111e9f0f0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))",),)

Same error. Any body can help please

**Comment by sroder** [comment]
for me, this issue is resolved.
My system was behind a company firewall and I had to set proxy server in .condarc file

**Comment by VinodRaj531** [comment]
@sroder Can you please tell me how you could solve this issue? Even I have strong company firewall and I guess that is giving me this error.

**Comment by sroder** [comment]
@Viraj531 :  I updated the .condarc file with below content

proxy_servers:
  http: http://<proxy-server>:port
  https: https://<prox-server>:port

**Comment by layibiyi** [comment]
Some might need to add indent when listing proxy_servers in the .condarc file. Didn't work for me till i did that i.e.

![condarc](https://user-images.githubusercontent.com/39195765/41311455-53ff98d2-6e52-11e8-9aab-0c1eb6cd6f4a.png)

**Comment by layibiyi** [comment]
Some might need to add indent when listing proxy_servers in the .condarc file. Didn't work for me till i did that i.e.

proxy_servers:
&nbsp;&nbsp;&nbsp;&nbsp; http: http://<proxy_server>:port
&nbsp;&nbsp;&nbsp;&nbsp; https: https://<proxy_server>:port

**Comment by NaveenKumarStark** [comment]
Thanks GUys

**Comment by samgonzalezz** [comment]
I just downloaded Anaconda with Anaconda Navigator 1.8.7 and it seems that by default the 'Enable SSL Verification' was activated. Under Anaconda Navigator Preferences I went and unchecked the box and then I was able to install packages using 'conda install'

**Comment by soumenms2015** [comment]
hello @samgonzalezz , I have unchecked the "Enable SSL Verification" But I am not still able to run conda update conda inside anaconda prompt. Are you able to install any module inside anaconda prompt? Can you please little elaborate? did you also configure .condarc file ?

**Comment by Saranya411** [comment]
Hi All,

We are getting the below issue when using conda install. Can anyone please help on this. We are at 4th week and still unable to fix it.

Solving environment: failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/free/linux-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://www.anaconda.com blocked, please file
a support request with your network engineering team.

ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/free/linux-64/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7fb3d9ecef28>: Failed to establish a new connection: [Errno 101] Network is unreachable',))",),)

**Comment by mirekphd** [comment]
I'm having the same problem during my docker build:
 
`ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/main/noarch/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f65d800b6d8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))",),)
`

I would push to docker hub the erroring image (before restarting the multi-hour build...) to help the Anaconda team reproduce the issue, but since the build did not complete, the container is inaccessible (I cannot docker run the image to docker exec _bash_ inside it). 

I hope that the usual method of cleaning the docker build cache by adding --no-cache to docker build solves the problem for me.

**Comment by sssubik** [comment]
Do we just write<proxy_server> or do we need to replace with certain values in the code below?

proxy_servers:
     http: http://<proxy_server>:port
     https: https://<proxy_server>:port

**Comment by Saranya411** [comment]
You need to give your proxy server..

**Comment by sssubik** [comment]
How do we get the proxy server? Is it only applicable for companies?

**Comment by adrian-moisa** [comment]
On windows 10 use "Anaconda Prompt"

**Comment by yogeshsaswade** [comment]
I am getting this error while installing: conda install -c conda-forge opentsne
Error:
ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/r/noarch/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00000000038E5F98>: Failed to establish a new connection: [Errno 10013] An attempt was made to access a socket in a way forbidden by its access permissions',))",),) 
How to solve this?

**Comment by kiciusKit** [comment]
In my case works fallowing: I open Anaconda Navigator, went to Enviroments tab. Click Channels bottom. Delete all channels. And add conda-forge

**Comment by mjahanbin** [comment]
I made a change to my PATH environment variable and it all started working after that. The change was to bring all anaconda related paths to beginning of the path. The reason this worked, in my case, was because "Intel\Intel(R) Management Engine Components\iCLS\" path, which originally came before the anaconda paths, was somehow interfering with conda operations. 

Hope this helps.

**Comment by aparnapai7** [comment]
Hi I'm facing the same exact issue - tried all the above steps and yet it throws the same error
Can someone suggest a solution?

**Comment by kalyc** [comment]
Hello, I am facing the same issue - Could someone please post an update here?
`
V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V

CondaHTTPError: HTTP 502 BAD GATEWAY for url <https://conda.anaconda.org/pytorch/linux-64/repodata.json>                                                                            
Elapsed: 00:08.168049
CF-RAY: 4a8b9cd23ff995aa-IAD

A remote server error occurred when trying to retrieve this URL.

A 500-type error (e.g. 500, 501, 502, 503, etc.) indicates the server failed to
fulfill a valid request.  The problem may be spurious, and will resolve itself if you
try your request again.  If the problem persists, consider notifying the maintainer
of the remote server.

A reportable application error has occurred. Conda has prepared the above report.
Upload successful.

`

**Comment by simest1989** [comment]
same issue here, this only started happening recently for me....

**Comment by kiransomase** [comment]
Hello I am also getting same error

V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://conda.anaconda.org/conda-forge/noarch/repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectionError(MaxRetryError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Max retries exceeded with url: /conda-forge/noarch/repodata.json (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x04B49B50>: Failed to establish a new connection: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions',))",),)

A reportable application error has occurred. Conda has prepared the above report.
Upload did not complete.

**Comment by shaikhalvee** [comment]
I am facing the same issue too.
My error was....

> PS C:\Windows\system32> conda update anaconda-navigator
Collecting package metadata: failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/msys2/win-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://www.anaconda.com blocked, please file
a support request with your network engineering team.

SSLError(MaxRetryError('HTTPSConnectionPool(host=\'repo.anaconda.com\', port=443): Max retries exceeded with url: /pkgs/msys2/win-64/repodata.json.bz2 (Caused by SSLError("Can\'t connect to HTTPS URL because the SSL module is not available."))'))

Then I stopped ssl verification.

> PS C:\Windows\system32> conda config --set ssl_verify False

But still getting the same error

> PS C:\Windows\system32> conda update anaconda-navigator
Collecting package metadata: failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/r/win-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://www.anaconda.com blocked, please file
a support request with your network engineering team.

SSLError(MaxRetryError('HTTPSConnectionPool(host=\'repo.anaconda.com\', port=443): Max retries exceeded with url: /pkgs/r/win-64/repodata.json.bz2 (Caused by SSLError("Can\'t connect to HTTPS URL because the SSL module is not available."))'))

I know I am not behind any proxy server. Can anyone solve it please

**Comment by shaikhalvee** [comment]
I resolved my problem by updating the anaconda path in environment variable.
In my path,
%CONDA_HOME% was set to /path/to/Anaconda3
And "%CONDA_HOME%" as well as "%CONDA_HOME%\Scripts" was added before.
To overcome the error, I had to add "%CONDA_HOME%\Library\bin" to path.
Now it works perfectly

**Comment by simest1989** [comment]
Thanks, can you please expand step by step how this was done? Thanks!

**Comment by keumarko** [comment]
Thanks shaikhalvee that worked perfectly for me too! 
simest 1989: here's what i did

Search for Anaconda3 in your folders, Open file location and go into Library then bin. Click a random file there and select copy file location in the toolbar. 
Go to "This PC" right-click, Properties, Advanced System Settings, Edit (System Variables), New.
Paste the location you copied in and delete the last file name in the address. Mine was:
"C:\Users\Mark\Anaconda3\Library\bin\xz.exe" but once last filename is deleted should look like
"C:\Users\Mark\Anaconda3\Library\bin\"

Hopefully that should help you!

**Comment by simest1989** [comment]
thanks for that clarification - this is now what i've done in sys variables:
name: python
value: C:\Users\me\Anaconda3\bin

However i'm still getting error when i try install_keras(tensorflow = 'gpu')

I won't post the entire error report but the following is the issue:
Failed to parse: proxy.server.com:port

Has anyone encountered this before and know how to get past this?

**Comment by shahardo** [comment]
I missed library/bin in the PATH. Added:
`C:\ProgramData\Anaconda3\Scripts
C:\ProgramData\Anaconda3\Library\bin`
and the problem was fixed

**Comment by simest1989** [comment]
figured it out, it's because i did an environment update in python that messed it up - once that was reset it started working!

**Comment by ratanGit** [comment]
I had the same HTTPs error. I think it was the missing environmental variable:

**%programdata%\Anaconda3\library\bin**

FYI:
(_%programdata% is the environmental variable for C:\Programdata_)

As I figured it out I would need 3 environmental variables on my Windows 10:

> C:\ProgramData\anaconda3\;C:\ProgramData\anaconda3\scripts\;C:\ProgramData\Anaconda3\Library\bin;
> 
![conda](https://user-images.githubusercontent.com/8282974/53704178-5cca9c80-3de8-11e9-84b6-c3cfaacd6065.png)

**Comment by TerryBryant** [comment]
Guys !!!
Finally I find the solution !!! Only for linux system !!!
You should find a  "libcrypto.so.1.0.0" file, and put it into your {anaconda_root}/lib, then create a soft link

`ln -s libcrypto.so.1.0.0 libcrypto.so`

For me, I find a "libcrypto.so.1.0.0" file in my anaconda virtural envs, and copy it to the destination.
I came across this problem just because I delete this .so file accidentally...

**Comment by BradXiao** [comment]
Try installing OpenSSL.
https://slproweb.com/products/Win32OpenSSL.html

**Comment by supratimdey** [comment]
just adding all 3 paths didn;t  work but installing the Openssl worked for me

**Comment by Elymar-Silva** [comment]
installing the Openssl worked for me too

**Comment by amansaini564** [comment]
1. Just untick the "Enable SSL Verification" in Preferences(Ctrl+P).
2. Update Index of root environment(Environment->base(root)->Update Index).
This worked for me. 
![SS](https://user-images.githubusercontent.com/30700306/56081088-e5e0e480-5e26-11e9-9dad-6ea49074d06c.jpg)

**Comment by wingmor99** [comment]
Installing the Openssl worked for me too.

**Comment by kartirai** [comment]
installing OpenSSL worked for me. Thanks a ton!

**Comment by kalyc** [comment]
How did you install openssl? From pip?

**Comment by kartirai** [comment]
Just install the software when you google search openSSL from the first link

Regards,

Kartikey Rai.

**Comment by NandhanThiravia** [comment]
Here is the solution for the HTTP Error and the SSL Error:
1) Add the proxy details in .condarc
2) Disable SSL Verify

.condarc file content
------------------------
ssl_verify: False
channels:
  - defaults
proxy_servers:
    http: http://<uname:passwd>@<ip:port>
    https: https://<uname:passwd>@<ip:port>

**Comment by MigVega** [comment]
My solution was the following:
1. Add the environmental variable:
%programdata%\Anaconda3\Library\bin
as @ratanGit mentioned before: https://github.com/ContinuumIO/anaconda-issues/issues/2760#issuecomment-469081254

2. Start Anaconda Prompt as Administrator (click right button of mouse before starting it), and then write:
`conda update conda`

**Comment by rhartvig** [comment]
Fix working for me: Sudo!

Not working: conda update anaconda => HTTP 000 CONNECTION FAILED but not behind a proxy

Working: sudo conda update anaconda

¯\_(ツ)_/¯

**Comment by cuixing158** [comment]
> On windows 10 use "Anaconda Prompt"

Thanks!

**Comment by Jimachin** [comment]
What I did was copy and paste from a installation of conda without problem, like this (en) :
Lo que hice fue copiar y pegar desde una instalación de conda sin problemas, como esto (es):

ssl_verify: true
channels:
  - defaults
default_python:

and this work for me (en).
and this works for me (es).

🤦‍♂🙃

**Comment by ZadeThakur** [comment]
The solution to issue is:
1. go to your anaconda prompt and type the command "where conda".
2. Two paths should pop up after that. add both of those paths to the path environment variables.(if you have already added any one of those already as your path variable you don't have to do it again).
3. after you have both of these path as your path variable. this issue should be resolved.

**Comment by mohdshoaib04** [comment]
Adding Channel conda:forge fixed my issue.

**Comment by Phone2017** [comment]
I encountered this, too. Someone give this simple code and I tried. Success!
conda config --set ssl_verify false

original: https://www.jianshu.com/p/777c7b7a3f3d

**Comment by yutongD** [comment]
Installing the Openssl worked for me too. THANKS!
https://slproweb.com/products/Win32OpenSSL.html

**Comment by Akhilsharma51** [comment]
I had the same error and it turns out that I was using my employer network for internet which had proxy and firewall set. Go to internet explorer and chose Lan settings (Connections tab) and check box 'Automatically detect ' (something like that) and then disconnect from your company's network or whatever and use your personal cellphone hotspot or whatever network you have for personal use. Run command again in anaconda terminal 'conda install <pkg name>' and it will work as it worked for me.

**Comment by keyochali** [comment]
thank you all
you saved my life
it took me 12 hours to solve it
and even I don't know how it solved
anyways thanks

I added library/bin to the path
and downloaded and installed https://slproweb.com/products/Win32OpenSSL.html
I think that was how I did it 
ohhh what a day

**Comment by tanshu007** [comment]
Installing the latest OpenSSL from https://slproweb.com/products/Win32OpenSSL.html works for me.

**Comment by iamsubrata** [comment]
For me this worked with below steps:
1> conda config --set ssl_verify no
2> Open the anaconda prompt (run as Administrator)
3> Go to inside the Scripts path of your  Anaconda folder
4> run the below
     conda install spacy

**Comment by xixiU** [comment]
just install OpenSSL-Win64 as above said  do not fix my problem.**But** after installing OpenSSL-Win64,I copy libcrypto-1_1-x64.dll and libcrypto-1_1-x64.dll to replace those dile in C:\Users\x\Anaconda3\DLLs  and C:\Users\x\Anaconda3\Library\bin solved my problem

**Comment by EbunAlabi** [comment]
> On windows 10 use "Anaconda Prompt"

This worked for me.

**Comment by maphew** [comment]
Updating the base environment is what worked for me:

    conda update --prefix c:\tools\miniconda3 conda
